faif/python-patterns
# pull (#1)

* Fix PEP8 issues (BLACK IT!). Configure flake8

* typo

* Allow delegator to return delegate's properties

The current code does not allow the delegator to return the value of properties of the delegate but can only call its methods. Moreover if the method does not exist then `None` is returned rather than raising an exception.
This PR is for fixing the above.

* .

* Update README with Contributing guidelines

* Update .travis.yml

Remove python 3.3

* Fix #247: simplify __str__ in structural/mvc.py

* Fix #250: assert error in structural/flyweight.py

* Refactor chain pattern

* Refactor Template pattern

* Make handlers a bit different, rename Default to Fallback

* Rename compare to check_range

* Refactor Strategy

* Remove coveralls

* Move from nosetests to pytest

* fix failing test

* Test outputs starting with two visitor and strategy

* Skip these tests for python2 because redirect_stdout is not available

* Update docs with some examples from python ecosystem

* Change project structure

* remove 3.7 from CI, fix flake warnings

* Update links in readme

* Add main function and tests for output for most of behavioral patterns

* Travis config: test with 3.7, drop 3.4, 3.5

* Add mock to requirements (needed for py2)

* Add notes to visitor

* Fix flake8 check and related warnings

* remove file added by mistake

* Update contributing guides

* Update Mediator pattern

* Rename chain to chain_of_responsibility

* Add Videos section to README

* Doctest Template

* Doctest Adapter

* Doctest Memento

* fix tests

* fix tests

* Combine and refine dependency injection patterns

* Remove old injection examples, update readme

* Update pattern description

* Substitute facade example with a better one

* Separate chain_of_responsibility script per python version

* Separate registry per python version

* Ignore py2-specific tests for py3 and vice versa

* Exclude files for flake8 and pytest in the same way

* Separate blackboard per python version

* Separate flyweight per python version

* Remove redundant test

* Setup tox

* fix travis

* Install codecov only when needed
